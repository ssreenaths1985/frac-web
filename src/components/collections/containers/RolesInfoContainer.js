import React from "react";
import ColumnOne from "../views/positions/ColumnOne";
import ColumnTwo from "../views/roles/ColumnTwo";
import ColumnThree from "../views/roles/ColumnThree";
import HeaderSection from "./common/HeaderSection";

class RolesInfoContainer extends React.Component {
  _isMounted = false;
  constructor(props) {
    super(props);
    this.state = {
      rolesCurrentDept: "",
      clients: "",
      roomId: "",
      gotData: "",
      sendData: "",
      allSessions: ""
    };
    this.getRolesCurrentDept = this.getRolesCurrentDept.bind(this);
    this.getClients = this.getClients.bind(this);
    this.getRoomId = this.getRoomId.bind(this);
    this.gotMessage = this.gotMessage.bind(this);
    this.sendMessage = this.sendMessage.bind(this);
    this.getSessions = this.getSessions.bind(this);
  }

  componentDidMount() {
    this._isMounted = true;
  }

  componentWillUnmount() {
    this._isMounted = false;
    this.setState = (state, callback) => {
      return;
    };
  }

  getRolesCurrentDept = (value) => {
    this.setState({
      rolesCurrentDept: value,
    });
  };

  getClients = (list) => {
    this.setState({
      clients: list,
    });
  };

  getRoomId = (value) => {
    this.setState({
      roomId: value,
    });
  };

  gotMessage = (data) => {
    this.setState({
      gotData: data,
    });
  };

  sendMessage = (data) => {
    this.setState({
      sendData: data,
    });
  };

  getSessions = (data) => {
    this.setState({
      allSessions: data
    })
  }

  render() {
    return (
      <React.Fragment>
        <div className="col-xs-12 col-sm-12 col-md-12 col-lg-12">
          <HeaderSection
            {...this.props}
            getClients={this.getClients}
            gotData={this.gotMessage}
            sendData={this.state.sendData}
            sendRoomId={this.state.roomId}
            getSessions={this.getSessions}
          />
        </div>
        <div className="d-xl-flex d-lg-flex d-md-flex d-sm-none col-xs-12 col-sm-12 col-md-12 col-lg-12 custom-padding-1">
          <ColumnOne
            {...this.props}
            getDeptValue={(value) => this.getRolesCurrentDept(value)}
            clients={this.state.clients}
            allSessions={this.state.allSessions}
          />
          <ColumnTwo
            {...this.props}
            department={this.state.rolesCurrentDept}
            clients={this.state.clients}
            allSessions={this.state.allSessions}
            getRoomId={this.getRoomId}
          />
          {this.props.history.location.state &&
            (this.props.history.location.state.showColumnThree ||
              this.props.history.location.pathname.includes("0")) && (
              <ColumnThree
                {...this.props}
                department={this.state.rolesCurrentDept}
                clients={this.state.clients}
                getMessage={this.state.gotData}
                sendMessage={this.sendMessage}
                allSessions={this.state.allSessions}
              />
            )}
          {(!this.props.history.location.state ||
            (this.props.history.location.state &&
              !this.props.history.location.state.stayOn &&
              !this.props.history.location.state.showColumnThree &&
              this.props.history.location.pathname === "/collection-roles/") ||
            this.props.history.location.state.deleteItem) && (
            <div className="col-xs-12 col-sm-12 col-md-8 col-lg-8 col-xl-8 bordered custom-full-height-4 custom-body-bg">
              <div
                className="vertical-center d-none d-sm-block"
                id="emptyState"
              >
                <center>
                  <h1 className="pb-2">No role selected</h1>
                  <p className="col-7 pb-2">
                    Please select a role from the list to see its details here
                  </p>
                  <img
                    src="../../img/empty/roles_empty.svg"
                    alt="empty activity"
                  ></img>
                </center>
              </div>
            </div>
          )}
        </div>
      </React.Fragment>
    );
  }
}

export default RolesInfoContainer;
