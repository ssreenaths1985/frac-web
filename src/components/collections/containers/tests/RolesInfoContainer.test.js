import React from "react";
import { shallow } from "enzyme";
import RolesInfoContainer from "../RolesInfoContainer";
const message = "No role selected";

const rolesInfoContainerProps = {
  history: {
    location: {
      pathname: "/collections",
      search: "",
      hash: "",
      state: null,
      key: "zgfeb7",
    },
  },
  handleClick: () => {
    console.log("click");
  },
};

let wrapped = shallow(
  <RolesInfoContainer {...rolesInfoContainerProps}></RolesInfoContainer>
);

describe("RolesInfoContainer", () => {
  it("should render the RolesInfoContainer Component correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
  it("renders the RolesInfoContainer children", () => {
    expect(wrapped.find("h1").text()).toEqual(message);
  });
});
