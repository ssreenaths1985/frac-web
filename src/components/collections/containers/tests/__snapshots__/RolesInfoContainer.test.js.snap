// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RolesInfoContainer should render the RolesInfoContainer Component correctly 1`] = `
<Fragment>
  <div
    className="col-xs-12 col-sm-12 col-md-12 col-lg-12"
  >
    <HeaderSection
      handleClick={[Function]}
      history={
        Object {
          "location": Object {
            "hash": "",
            "key": "zgfeb7",
            "pathname": "/collections",
            "search": "",
            "state": null,
          },
        }
      }
    />
  </div>
  <div
    className="d-xl-flex d-lg-flex d-md-flex d-sm-none col-xs-12 col-sm-12 col-md-12 col-lg-12 custom-padding-1"
  >
    <ColumnOne
      getDeptValue={[Function]}
      handleClick={[Function]}
      history={
        Object {
          "location": Object {
            "hash": "",
            "key": "zgfeb7",
            "pathname": "/collections",
            "search": "",
            "state": null,
          },
        }
      }
    />
    <ColumnTwo
      department=""
      handleClick={[Function]}
      history={
        Object {
          "location": Object {
            "hash": "",
            "key": "zgfeb7",
            "pathname": "/collections",
            "search": "",
            "state": null,
          },
        }
      }
    />
    <div
      className="col-xs-12 col-sm-12 col-md-8 col-lg-8 col-xl-8 bordered custom-full-height-4 custom-body-bg"
    >
      <div
        className="vertical-center d-none d-sm-block"
        id="emptyState"
      >
        <center>
          <h1
            className="pb-2"
          >
            No role selected
          </h1>
          <p
            className="col-7 pb-2"
          >
            Please select a role from the list to see its details here
          </p>
          <img
            alt="empty activity"
            src="../../img/empty/roles_empty.svg"
          />
        </center>
      </div>
    </div>
  </div>
</Fragment>
`;
